features/timeseries/src/test/java/org/opennms/netmgt/timeseries/resource/TimeseriesResourceStorageDaoTest.java

```diff
@@ -222,26 +219,15 @@ private void replay() throws StorageException {
                 ResourcePath resourcePath = (ResourcePath)EasyMock.getCurrentArguments()[0];
                 int depth = (Integer)EasyMock.getCurrentArguments()[1];
 
-                int idxSuffix = resourcePath.elements().length - 1;
-                int targetLen = idxSuffix + depth + 2;
-                String field = "_idx"+idxSuffix; // key
-                String value = String.format("(%s,%d)", toResourceId(resourcePath), targetLen); // value
-
+                String regex = toSearchRegex(resourcePath, depth + 1);
                 Set<Metric> metrics = new HashSet<>();
                 for (Entry<ResourcePath, Set<String>> entry : indexedPaths.entrySet()) {
-                    Map<String, String> attributes = Maps.newHashMap();
-                    // Build the indexed attributes and attempt to match them against the given query
-                    TimeseriesUtils.addIndicesToAttributes(entry.getKey(), attributes);
-                    if (value.equals(attributes.get(field))) {
-
-                        for(String name : entry.getValue()) {
+                    if (toResourceId(entry.getKey()).matches(regex)) { // find all paths that match our regex
```

Comment:
Let's use RE2 for evaluating and matching any of the regular expressions: https://github.com/OpenNMS/opennms-integration-api/pull/42

Reply:
fixed.